src/map/utils/charutils.cpp

```diff
@@ -2063,13 +2063,17 @@ namespace charutils
                             case SKILL_KATANA:
                             case SKILL_CLUB:
                             {
-                                if (PItem->isType(ITEM_WEAPON) &&
-                                    (!charutils::hasTrait(PChar, TRAIT_DUAL_WIELD) || ((CItemWeapon*)PItem)->getSkillType() == SKILL_NONE))
+                                bool isWeapon = PItem->isType(ITEM_WEAPON);
+                                if (isWeapon && (!charutils::hasTrait(PChar, TRAIT_DUAL_WIELD) || ((CItemWeapon*)PItem)->getSkillType() == SKILL_NONE))
```

Comment:
This shouldn't be any different because `isType()` should already be returning truthy, and then the 2nd change you do below to shouldn't get hit because we should have already returned false here. Neither change should have been necessary which lead me to believe there is a further underlying problem, perhaps with the cast going on in `isType()`

Reply:
nope nope, I'm reading this wrong, it has to be weapon in both places - I was reading it like "not weapon" in the 1st check. never mind me, blond moment.